packages/server/lib/browsers/webkit.ts

```diff
@@ -26,22 +31,47 @@ export function connectToExisting () {
 export async function open (browser: Browser, url: string, options: BrowserLaunchOpts, automation: Automation): Promise<BrowserInstance> {
   // resolve pw from user's project path
   const pwModulePath = require.resolve('playwright-webkit', { paths: [process.cwd()] })
-  const pw = require(pwModulePath) as typeof playwright
+  const pw = await import(pwModulePath) as typeof playwright
 
-  const pwBrowser = await pw.webkit.launch({
-    proxy: {
-      server: options.proxyServer,
+  const defaultLaunchOptions = {
+    preferences: {
+      proxy: {
+        server: options.proxyServer,
+      },
+      headless: browser.isHeadless,
     },
-    downloadsPath: options.downloadsFolder,
-    headless: browser.isHeadless,
-  })
+    extensions: [],
+    args: [],
+  }
+
+  const launchOptions = await utils.executeBeforeBrowserLaunch(browser, defaultLaunchOptions, options)
+
+  if (launchOptions.extensions.length) options.onWarning?.(new Error('WebExtensions not supported in WebKit, but extensions were passed in before:browser:launch.'))
+
+  launchOptions.preferences.args = [...launchOptions.args, ...(launchOptions.preferences.args || [])]
+
+  const pwServer = await pw.webkit.launchServer(launchOptions.preferences)
+
+  /**
+   * Playwright adds an `exit` event listener to run a cleanup process. It tries to use the current binary to run a Node script by passing it as argv[1].
+   * However, the Electron binary does not support an entrypoint, leading Cypress to think it's being opened in global mode (no args) when this fn is called.
+   * Solution is to filter out the problematic function.
+   * TODO(webkit): do we want to run this cleanup script another way?
+   * @see https://github.com/microsoft/playwright/blob/7e2aec7454f596af452b51a2866e86370291ac8b/packages/playwright-core/src/utils/processLauncher.ts#L191-L203
```

Comment:
Ahh this must have been tricky to debug. Nice

Reply:
Super tricky. The stdout/stderr isn't inherited and it doesn't pass `--inspect-brk` so it was like a ghost process. I eventually figured out what was going on by adding logging using `fs.createWriteStream` and `tail`'ing the file.